### ISSUE TITLE
GVRKit plugin metadata generation issue

### ISSUE THREAD

**Issue by NathanaelA** [NativeScript/ios-runtime#877]
I have a GVRKit plugin that I have been developing for a project, and the runtime metadata is incorrect, and I need to find a solution to get around this in NativeScript.

In some ways slightly related to #826 (Like him I've got one that is working).
Partially related to this issue #710...

Error is:
`Error: Actual arguments count: "0". Expected: "1".`

```
      - Name:            handleTrigger
        JsName:          handleTrigger
        Filename:        GVRRenderer.h
        Module:          
          FullName:        GVRKit.GVRRenderer
          IsPartOfFramework: true
          IsSystemModule:  false
          Libraries:       
        Flags:           [ IsIosAppExtensionAvailable ]
        Type:            Method
        Signature:       
          - Type:            Bool

      - Name:            'handleTrigger:'
        JsName:          handleTrigger
        Filename:        GVRRenderer.h
        Module:          
          FullName:        GVRKit.GVRRenderer
          IsPartOfFramework: true
          IsSystemModule:  false
          Libraries:       
        Flags:           [ IsIosAppExtensionAvailable ]
        Type:            Method
        Signature:       
          - Type:            Bool
          - Type:            Interface
            JsName:          GVRHeadPose
            WithProtocols:   
```

Do you see that the metadata JS name is the SAME, but the core name is different.   One has a prototype of:
`bool handleTrigger()`
and the other  is
`bool handleTrigger(GVRHeadPose *headpose)`

I really want to call the first option.  I can't seem to because NativeScript has decided the only valid one is the second one.  ;-(

Solutions?

**Comment by ulvesked** [comment]
Hi, I have the same issues with StarIO-framework.

      - Name:            'appendLineFeed:'
        JsName:          appendLineFeed
        Filename:        /Users/anders/Apps/nativescript-star-io/demo/node_modules/nativescript-star-io/platforms/ios/StarIO_Extension.framework/Headers/ISCBBuilder.h
        Module:          
          FullName:        StarIO_Extension.ISCBBuilder
          IsPartOfFramework: true
          IsSystemModule:  false
          Libraries:       
        Flags:           [ IsIosAppExtensionAvailable ]
        Type:            Method
        Signature:       
          - Type:            Void
          - Type:            Int
      - Name:            appendLineFeed
        JsName:          appendLineFeed
        Filename:        /Users/anders/Apps/nativescript-star-io/demo/node_modules/nativescript-star-io/platforms/ios/StarIO_Extension.framework/Headers/ISCBBuilder.h
        Module:          
          FullName:        StarIO_Extension.ISCBBuilder
          IsPartOfFramework: true
          IsSystemModule:  false
          Libraries:       
        Flags:           [ IsIosAppExtensionAvailable ]
        Type:            Method
        Signature:       
          - Type:            Void

This is not the only place I have discovered it. It is the same with `appendMultiple:height:` vs `appendMultipleHeight:` that both is recognised as appendMultipleHeight in Nativescript.

Is there a plan to fix this?

**Comment by mbektchiev** [comment]
Hi @ulvesked. Thank you for the ideas presented. I like approach number 3 the most, but it has a small drawback - there could still be cases where the number of arguments will be the same. Consider the following hypothetical example:
```
- (void)appendMultiple:(NSInteger)width height:(NSInteger)height;
- (void)append:(NSInteger)height multipleHeight: (NSInteger)multipleHeight;
```
Both methods would end up with the same name (`appendMultipleHeight`) and the same number of arguments (two). That's why I think that besides the logic that you proposed, we could make the metadata generator fallback to appending a number suffix to the JS name of the method whenever it detects such a collision, which I suppose will be an extremely rare situation.

We will try to fix the issue for the upcoming 5.0 release of {N}.
